feature_examples/tensorflow2/pipelining/pipelining.py

```diff
@@ -0,0 +1,425 @@
+        x = Flatten(name='flatten')(input_layer)
+        x = Dense(256, activation='relu', name="dense256")(x)
+        x = Dense(128, activation='relu', name="dense128")(x)
+        x = Dense(64, activation='relu', name="dense64")(x)
+
+    with ipu.keras.PipelineStage(1):
+        x = Dense(32, activation='relu', name="dense32")(x)
+        x = Dense(10, name="logits")(x)
+
+    model = Model(inputs=input_layer,
+                  outputs=x,
+                  name="multipleIPUfunctional")
+    return model
+"""
+In case an existing `TensorFlow` model is imported, an additional API
+is provided to facilitate managing Pipeline Stages assignments.
+
+This feature is implemented with `model.get_pipeline_stage_assignment()`
+and `model.set_pipeline_stage_assignment(assignments)` where `assignments` is
+the result of calling `get_pipeline_stage_assignment`.
+For an example with the ResNet50 please check this [documentation](https://docs.graphcore.ai/projects/tensorflow-user-guide/en/latest/keras_tf2.html#pipelining-an-existing-functional-model).
+"""
+"""
+### Execute Training
+
+Next we refresh IPU device configuration and train again with the new model.
+"""
+ipu_configuration = make_ipu_config(num_ipus=2)
+
+train(
+    strategy=ipu.ipu_strategy.IPUStrategy(),
+    model_factory=create_functional_model_with_stages,
+    train_ds=train_ds
+)
+# sst_hide_output
+"""
+##  Training a Keras `Sequential model` with pipelining
+
+Next we will write a function to create the model using the Keras `Sequential`
+class as we did above, but with explicit mapping of layers to stages through 
+`set_pipeline_stage_assignment`, which accepts a list of integers as
+a parameter. This function sets the pipeline stage assignment for all
+the invocations of all the layers (excluding input layers) in the model
+which is used to create a model-parallel execution when calling `fit()`,
+`evaluate()` and `predict()`. 
+
+>This pipelining stage assignment is ignored when using the `call()` function
+>on this model.
+
+Below you will see pipeline stage assignment like this: `[0, 0, 1, 1 1, 1])`. 
+This means that first two layers of `Sequential` model are assigned to 
+the first stage, and the remaining layers to the second stage.
+
+This list has to be has to be of the same length as the total number
+of invocations of all the layers in this model, excluding input layers.
+
+### Define the model
+"""
+def create_pipeline_sequential_model():
+    seq_model = Sequential(
+        layers=[
+            Flatten(name='flatten'),
+            Dense(256, activation='relu', name="dense256"),
+            Dense(128, activation='relu', name="dense128"),
+            Dense(64, activation='relu', name="dense64"),
+            Dense(32, activation='relu', name="dense32"),
+            Dense(10, activation='softmax', name="logits")
+        ],
+        name="multipleIPUsequential"
+    )
+    seq_model.set_pipeline_stage_assignment([0, 0, 1, 1, 1, 1])
```

Comment:
Wouldn't it be more consistent if the division between stages was the same as in the earlier stage? I mean
```suggestion
    seq_model.set_pipeline_stage_assignment([0, 0, 0, 0, 1, 1])
```

Reply:
If yes, there are more places tu update by the way :P

Reply:
nice catch.